crates/types/src/lib.rs

```diff
@@ -69,8 +69,117 @@ impl fmt::Display for WasmType {
             WasmType::F32 => write!(f, "f32"),
             WasmType::F64 => write!(f, "f64"),
             WasmType::V128 => write!(f, "v128"),
-            WasmType::ExternRef => write!(f, "externref"),
-            WasmType::FuncRef => write!(f, "funcref"),
+            WasmType::Ref(rt) => write!(f, "{}", rt),
+            WasmType::Bot => write!(f, "bot"),
+        }
+    }
+}
+
+/// WebAssembly reference type -- equivalent of `wasmparser`'s RefType
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct WasmRefType {
+    pub nullable: bool,
+    pub heap_type: WasmHeapType,
+}
+
+pub const WASM_EXTERN_REF: WasmRefType = WasmRefType {
+    nullable: true,
+    heap_type: WasmHeapType::Extern,
+};
+
+pub const WASM_FUNC_REF: WasmRefType = WasmRefType {
+    nullable: true,
+    heap_type: WasmHeapType::Func,
+};
```

Comment:
In retrospect, I think this might be better modeled as perhaps:

```rust
impl ValType {
    pub const EXTERNREF = ...;
}

impl WasmRefType {
    pub const EXTERN: WasmRefType = ...;
}
```

basically just using associated `const`s to namespace things instead of splatting it into the module namespace here.